lib/kubeclient/common.rb

```diff
@@ -229,6 +229,31 @@ def retrieve_all_entities(entity_types)
       end
     end
 
+    def get_pod_log(pod_name, namespace, container: nil, previous: false)
+      params = {}
+      params[:previous] = true if previous
+      params[:container] = container if container
+
+      ns = build_namespace_prefix(namespace)
+      handle_exception do
+        rest_client[ns + "pods/#{pod_name}/log"]
+          .get({ 'params' => params }.merge(@headers))
+      end
+    end
+
+    def watch_pod_log(pod_name, namespace, container: nil)
+      params = { follow: true }
```

Comment:
Yes, that's a Kubernetes API issue.  Notice that the endpoint for `get_pod_logs` and `watch_pod_logs` is the same. The only difference is for watch, we're adding the `follow=true` param.  If you check out the [API Docs](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/HEAD/docs/api-reference/v1/operations.html) and search for "/log", you'll see the parameters it accepts.

The Kubernetes team is planning to remove the /watch/* endpoints entirely, and use a query param instead.  (See [PR 8337](https://github.com/kubernetes/kubernetes/issues/8337).)  But I'll open up a separate PR to deal with that issue.